neuro_san/http_sidecar/interfaces/agent_authorizer.py

```diff
@@ -20,10 +21,10 @@ class AgentAuthorizer:
     of allowing to route incoming requests to an agent.
     """
 
-    def allow(self, agent_name) -> bool:
+    def allow(self, agent_name) -> AsyncAgentService:
         """
```

Comment:
Instead of just saying "Yes/No" for agent request authorization,
return an instance of AsyncAgentService for this agent if "Yes",
None for "No".